README.md

```diff
@@ -1,20 +1,94 @@
+# Description
+
+[libp2p](https://github.com/ipfs/specs/tree/master/libp2p) is a networking stack and library modularized out of [The IPFS Project](https://github.com/ipfs/ipfs), and bundled separately for other tools to use.
 >
-> libp2p is the product of a long, and arduous quest of understanding -- a deep dive into the internet's network stack, and plentiful peer-to-peer protocols from the past. Building large scale peer-to-peer systems has been complex and difficult in the last 15 years, and libp2p is a way to fix that. It is a "network stack" -- a protocol suite -- that cleanly separates concerns, and enables sophisticated applications to only use the protocols they absolutely need, without giving up interoperability and upgradeability. libp2p grew out of IPFS, but it is built so that lots of people can use it, for lots of different projects.
+libp2p is the product of a long, and arduous quest of understanding -- a deep dive into the internet's network stack, and plentiful peer-to-peer protocols from the past. Building large scale peer-to-peer systems has been complex and difficult in the last 15 years, and libp2p is a way to fix that. It is a "network stack" -- a protocol suite -- that cleanly separates concerns, and enables sophisticated applications to only use the protocols they absolutely need, without giving up interoperability and upgradeability. libp2p grew out of IPFS, but it is built so that lots of people can use it, for lots of different projects.
 >
 > We will be writing a set of docs, posts, tutorials, and talks to explain what p2p is, why it is tremendously useful, and how it can help your existing and new projects. But in the meantime, check out
 >
 > - [**The IPFS Network Spec**](https://github.com/ipfs/specs/tree/master/protocol/network), which grew into libp2p
 > - [**go-libp2p implementation**](https://github.com/ipfs/go-libp2p)
-> - [**node-libp2p implementation**](https://github.com/diasdavid/node-libp2p)
+> - [**js-libp2p implementation**](https://github.com/diasdavid/js-libp2p)
+
+# Contribute
+
+libp2p implementation in Go is a work in progress. As such, there's a few things you can do right now to help out:
+ - Go through the modules below and **check out existing issues**. This would be especially useful for modules in active development. Some knowledge of IPFS/libp2p may be required, as well as the infrasture behind it - for instance, you may need to read up on p2p and more complex operations like muxing to be able to help technically.
+ - **Perform code reviews**. 
+ - **Add tests**. There can never be enough tests.
+
+# Usage
+
+`go-libp2p` repo will be a place holder for the list of Go modules that compose Go libp2p, as well as its entry point.
+
+## Install
+
+```bash
+$ go get github.com/ipfs/go-libp2p
+```
+
+# Run tests
+
+```bash
+$ cd $GOPATH/src/github.com/ipfs/go-libp2p
+$ GO15VENDOREXPERIMENT=1 go test ./p2p/<path of _test.go you want to run>
```

Comment:
just advise people to use the path to the package they want to test. the `-run` flag can select individual tests from there